pkg/config/validation/validation.go

```diff
@@ -228,6 +230,25 @@ func ValidateHTTPHeaderName(name string) error {
 	if name == "" {
 		return fmt.Errorf("header name cannot be empty")
 	}
+	if !validHeaderRegex.MatchString(name) {
+		return fmt.Errorf("header name %s is not a valid header name", name)
+	}
+	return nil
+}
+
+// ValidateHTTPHeaderNameOrJwtClaimRoute validates a header name, allowing special @request.auth.claims syntax
+func ValidateHTTPHeaderNameOrJwtClaimRoute(name string) error {
+	if name == "" {
+		return fmt.Errorf("header name cannot be empty")
+	}
+	if jwt.ToRoutingClaim(name).Match {
```

Comment:
Should we cut the `@request.auth.claims` prefix and then validate according to the rules of the header?